.github/workflows/run-composition.yml

```diff
@@ -28,10 +28,31 @@ on:
       test_ref:
         required: false
         type: string
+
 jobs:
+  choose_runner:
+    runs-on: ubuntu-latest
+    outputs:
+      config: ${{ steps.config.outputs.result }}
+    steps:
+      - uses: actions/github-script@v6
+        id: config
+        with:
+          script: |
+            if (`${context.repo.owner}/${context.repo.repo}` === 'libp2p/test-plans') {
```

Comment:
That approach looks great. Do we need to take into account the fact that these jobs are used through reusable workflows?

https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-github-hosted-runners

Reply:
If I understand the docs correctly, the context here will reference the `caller` repo which is exactly what we want. So if we call the reusable workflow from `libp2p/test-plans`, then we're going to use self-hosted runners. Otherwise, we'll fall back to the gh hosted ones.